src/components/Carousel.test.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Carousel from './Carousel';
+
+describe('<Carousel/>', () => {
+  let wrapper, props;
+
+  beforeEach(() => {
+    props = {
+      images: [
+        {
+          id: '123yvu1231bv23',
+          url: 'https://placeholder.com/480x200.png',
+          description: 'brief image description here'
+        },
+        {
+          id: '09nhgfnbdv212j',
+          url: 'https://placeholder.com/480x250.png',
+          description: 'brief image description here'
+        }
+      ]
+    };
+    wrapper = shallow(<Carousel {...props} />);
+  });
+  it('renders all images and only once', () => {
+    let renderedImages;
+    props.images.forEach(image => {
+      renderedImages = wrapper
+        .findWhere(child => {
+          return (
+            child.type() === 'img' &&
+            child.prop('src') === image.url &&
+            child.prop('alt') === image.description
+          );
+        })
+        .map(renderedImage => renderedImage.prop('src'));
+      expect(renderedImages).toContain(image.url);
+      expect(renderedImages.length).toBe(1);
+    });
+  });
+  it("doesn't render if props are not provided", () => {
```

Comment:
take a look at [prop types](https://reactjs.org/docs/typechecking-with-proptypes.html) That could eliminate two of these tests, checking if props are provided and checking if they're the correctd type.

Reply:
https://github.com/CodeForBoulder/upswyng/pull/47

sell proptypes >> buy typescript

Reply:
I'll leave this as is until we decide which direction to go. We'll discuss on #47